File: app/src/main/java/org/magnum/logger/connectivity/wifi/WifiTable.java
package org.magnum.logger.connectivity.wifi;

import org.magnum.logger.Database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

public class WifiTable extends Database
{

	private static final String TABLE_NAME = "WiFi";
	private static final String COLUMN_NAMES[] = { "Time", "BSSID" };

	public WifiTable(Context context)
	{
		super(context);
	}

	public static void create(SQLiteDatabase db)
	{
		db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_NAME + "( Time INTEGER NOT NULL,  BSSID INTEGER NOT NULL )");
	}

	@Override
	public String[] getColumnNames()
	{
		return COLUMN_NAMES;
	}

	@Override
	public String getTableName()
	{
		return TABLE_NAME;
	}
}

File: app/src/main/java/org/magnum/logger/connectivity/wifi/WifiTable.kt
package org.magnum.logger.connectivity.wifi

import androidx.room.Dao
import androidx.room.Entity
import androidx.room.Insert
import androidx.room.PrimaryKey

const val NAME = "Wifi"

@Entity(tableName = NAME)
data class WifiRecord(@PrimaryKey val time: Long, val bssid: String)

@Dao
interface WifiTable {
    @Insert
    fun insert(vararg record: WifiRecord)
}